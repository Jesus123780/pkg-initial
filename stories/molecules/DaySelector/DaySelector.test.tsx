import { fireEvent, render } from '@testing-library/react'
import React from 'react'
import { DaySelector } from './index'

describe('DaySelector', () => {
  const days = [
    { day: 1, name: 'Monday' },
    { day: 2, name: 'Tuesday' }
    // ... add more days as needed
  ]

  it('renders correctly', () => {
    const { container } = render(<DaySelector
      days={days}
      handleDaySelection={() => {}}
      selectedDays={[1]}
    />)
    expect(container).toMatchSnapshot()
  })

  it('calls handleDaySelection on circle click', () => {
    const handleDaySelection = jest.fn()
    const { getByText } = render(<DaySelector
      days={days}
      handleDaySelection={handleDaySelection}
      selectedDays={[1]}
    />)

    fireEvent.click(getByText('Monday')) // Assuming 'Monday' is in the list

    expect(handleDaySelection).toHaveBeenCalledWith(1)
  })
})
