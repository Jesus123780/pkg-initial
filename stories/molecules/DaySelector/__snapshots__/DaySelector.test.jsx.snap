// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaySelector renders correctly 1`] = `
<div>
  <div
    class="sc-bcXHqe ioEsdV"
  >
    Monday
  </div>
  <div
    class="sc-bcXHqe TXgoR"
  >
    Tuesday
  </div>
</div>
`;
